modules/worker/dataset-pruning-worker.js

```diff
@@ -0,0 +1,30 @@
+const logger = require('../../modules/logger');
+const GraphStorage = require('../../modules/Database/GraphStorage');
+const DatasetPruningService = require('../../modules/service/dataset-pruning-service');
+
+process.on('message', async (data) => {
+    const {
+        selectedDatabase,
+        importedPruningDelayInMinutes,
+        replicatedPruningDelayInMinutes,
+        repackedDatasets,
+    } = JSON.parse(data);
+    try {
+        const graphStorage = new GraphStorage(selectedDatabase, logger);
+        const datasetPruningService = new DatasetPruningService({ logger, graphStorage });
+        const idsForPruning = await datasetPruningService
+            .getIdsForPruning(
+                repackedDatasets,
+                importedPruningDelayInMinutes,
+                replicatedPruningDelayInMinutes,
+            );
+        if (idsForPruning.datasetsToBeDeleted.length === 0) {
+            process.send(idsForPruning);
```

Comment:
The return statement is the same so could we exclude it from the if statement?

Reply:
Good catch, i've updated the code